[BZ: 2004921][MULTIARCH:1392][WIP]: Updates for IBM Power OCP 4.9 (#36418)
openshift/openshift-docs: modules/rhcos-enabling-multipath-day-1-power.adoc

// Module included in the following assemblies:
//
// * installing/installing_bibm_power/installing-ibm-power.adoc
// * installing/installing_ibm_power/installing-restricted-networks-ibm-power.adoc

[id="rhcos-enabling-multipathday-1-power_{context}"]
= Enabling multipathing with kernel arguments on {op-system}

You can enable multipathing at installation time for nodes that were provisioned in {product-title} 4.9 or later. {op-system} supports multipathing on the primary disk, allowing stronger resilience to hardware failure to achieve higher host availability. Although it is often preferable to modify kernel arguments as a day-2 activity,
you might want to add kernel arguments to all master or worker nodes during initial cluster
installation via the machine config.

> To add kernel arguments to master or worker nodes, you can create a `MachineConfig` object

Reviewer: ```suggestion
During the initial cluster creation, you might want to add kernel arguments to all master or worker nodes. To add kernel arguments to master or worker nodes, you can create a `MachineConfig` object and inject that object into the set of manifest files used by Ignition during cluster setup. 
```

I would also include a link to where the procedure is for adding kernel arguments. This way you can drop the step to "decide" below.

and inject that object into the set of manifest files used by Ignition during
cluster setup.

.Procedure

. Change to the directory that contains the installation program and generate the Kubernetes manifests for the cluster:
+
[source,terminal]
----
$ ./openshift-install create manifests --dir=<installation_directory>
----

. Decide if you want to add kernel arguments to worker or master nodes.

> * Create a machine config file, such as `99-master-kargs-mpath.yaml`, that instructs the cluster to add the `master` label and that identifies the multipath kernel argument, for example:

Reviewer: ```suggestion
* Create a machine config file. For example, create a `99-master-kargs-mpath.yaml` that instructs the cluster to add the `master` label and identify the multipath kernel argument:
```

+
[source,yaml]
----
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: "master"
  name: 99-master-kargs-mpath
spec:
  kernelArguments:
    - 'rd.multipath=default'
    - 'root=/dev/disk/by-label/dm-mpath-root'
----

. To enable multipathing post-installation on worker nodes:

* Create a machine config file, such as `99-worker-kargs-mpath.yaml`, that instructs the cluster to add the `worker` label and that identifies the multipath kernel argument, for example:
+
[source,yaml]
----
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: "worker"
  name: 99-worker-kargs-mpath
spec:
  kernelArguments:
    - 'rd.multipath=default'
    - 'root=/dev/disk/by-label/dm-mpath-root'
----
+

> You can change `master` to `worker` to add kernel arguments to worker nodes instead.

Reviewer: I don't think this is needed. You just showed us how to add worker nodes. I would remove this sentence.

> Create a separate YAML file to add to both master and worker nodes.

Reviewer: Same statement here. If you want to keep this statement, I would add this before the instructions about worker and master nodes.

> You can now continue on to create the cluster.

Reviewer: Not sure the value this statement adds.

[IMPORTANT]
====
Additional post-installation steps are required to fully enable multipathing. For more information, see “Enabling multipathing with kernel arguments on {op-system}" in _Post-installation machine configuration tasks_.
====

> In case of MPIO failure, following bootlist command can be used to update the boot device list with alternate logical device names.

Reviewer: ```suggestion
In case of MPIO failure, use the bootlist command to update the boot device list with alternate logical device names.
```
